src/clean-core/native/memory.hh

```diff
@@ -0,0 +1,19 @@
+#pragma once
```

Comment:
I'd like to discuss a owning RAII version of these and if we can a bit of impedance to the "raw" functions (as they are so easy to use wrongly. maybe move in a sub namespace or something).

Reply:
These functions are used by the vmem allocators, so convenient usage like that should just be done via them and existing container types. I was thinking about a namespace as well but this is already in the native/ folder